### PR TITLE
feat: adding PACHD_ADDRESS and DEX_TOKEN to task env [DET-9875]

### PR DESCRIPTION
**Pull request by salonig23**
## Description

Added PACHD_ADDRESS to the task environment variables as a part of pachyderm user provisioning.

## Test Plan

1. Create a Kubernetes cluster using the determined-ee repo.
2. Checkout branch saloni-gupta-pachyderm-intg.
3. Add your pachyderm address to the `./devcluster.yaml`:
```
integrations:
  pachyderm:
    address: "foo"
```
4. Start the devcluster
5. Login using your OIDC credentials.
6. Open a jupyter notebook.
7.  Run the command `!env` in the notebook.
8. Verify that the PACHD_ADDRESS environment variable is the same as the one you put in the `./devcluster.yaml` file.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/internal/api_command.go [hunk 1]
```diff
@@ -402,10 +417,21 @@ func (a *apiServer) LaunchCommand(
...
+
 	launchReq.Spec.Base.ExtraEnvVars = map[string]string{
 		"DET_TASK_TYPE": string(model.TaskTypeCommand),
 	}
 
+	OIDCPachydermEnvVars, err := a.AddOIDCPachydermEnvVars(session)
+	if err != nil {
+		return nil, err
+	}
+	if len(OIDCPachydermEnvVars) == 0 {
```
[the remark this answers is not in the document]
Comment (salonig23) [reply master/internal/api_command.go]: done

File: master/internal/api_command.go [hunk 2]
```diff
@@ -352,10 +352,25 @@ func (a *apiServer) SetCommandPriority(
 	return &apiv1.SetCommandPriorityResponse{Command: cmd.ToV1Command()}, nil
 }
 
+func (a *apiServer) AddOIDCPachydermEnvVars(
```
[the remark this answers is not in the document]
Comment (salonig23) [reply master/internal/api_command.go]: done

**Comment by stoksc** [comment]
lgtm modulo some style comments

File: master/internal/api_command.go [hunk 3]
```diff
@@ -352,10 +352,25 @@ func (a *apiServer) SetCommandPriority(
 	return &apiv1.SetCommandPriorityResponse{Command: cmd.ToV1Command()}, nil
 }
 
+func (a *apiServer) GetOIDCPachydermEnvVars(
```
[the remark this answers is not in the document]
Comment (salonig23) [reply master/internal/api_command.go]: done

**Pull request merged**